### PR TITLE
Reduce disk space usage when packaging the built SDK

### PR DESCRIPTION
**Pull request by jonsimantov**
Remove intermediate build files during desktop packaging step.
***

Packaging workflow run.
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [ ] Other, such as a build process or documentation change.
***

#### Notes
- Bug fixes and feature changes require an update to the `Release Notes` section of `release_build_files/readme.md`.
- Read the contribution guidelines [CONTRIBUTING.md](https://github.com/firebase/firebase-cpp-sdk/blob/main/CONTRIBUTING.md).
- Changes to the public API require an internal API review. If you'd like to help us make Firebase APIs better, please propose your change in a feature request so that we can discuss it together.

### CODE REVIEW

**Review by a-maurice — APPROVED** [review]

File: .github/workflows/cpp-packaging.yml [hunk 1]
```diff
@@ -433,6 +433,8 @@ jobs:
         run: |
           cd out-sdk
           find .. -type f -print > src_file_list.txt
+          # Remove intermediate build files (.o and .obj) files to save space.
+          find . -type f -name '*.o' -or -name '*.obj' -print0 | xargs -0 rm -f --
```
Comment (dconeybe) [inline .github/workflows/cpp-packaging.yml:437]: Drive-by comment: Would it make sense to run `df` before and after this command so you could quantify how much disk space was actually saved?

Comment (jonsimantov) [reply .github/workflows/cpp-packaging.yml:437]: Yeah that probably would have been a good idea. :)
